config/cloudinit/datasource/metadata/digitalocean/metadata.go

```diff
@@ -80,28 +86,83 @@ func (ms *MetadataService) FetchMetadata() (metadata datasource.Metadata, err er
 		return
 	}
 
-	if len(m.Interfaces.Public) > 0 {
-		if m.Interfaces.Public[0].IPv4 != nil {
-			metadata.PublicIPv4 = net.ParseIP(m.Interfaces.Public[0].IPv4.IPAddress)
+	/*
+		if len(m.Interfaces.Public) > 0 {
+			if m.Interfaces.Public[0].IPv4 != nil {
+				metadata.PublicIPv4 = net.ParseIP(m.Interfaces.Public[0].IPv4.IPAddress)
+			}
+			if m.Interfaces.Public[0].IPv6 != nil {
+				metadata.PublicIPv6 = net.ParseIP(m.Interfaces.Public[0].IPv6.IPAddress)
+			}
+		}
+		if len(m.Interfaces.Private) > 0 {
+			if m.Interfaces.Private[0].IPv4 != nil {
+				metadata.PrivateIPv4 = net.ParseIP(m.Interfaces.Private[0].IPv4.IPAddress)
+			}
+			if m.Interfaces.Private[0].IPv6 != nil {
+				metadata.PrivateIPv6 = net.ParseIP(m.Interfaces.Private[0].IPv6.IPAddress)
+			}
+		}
+	*/
+	metadata.NetworkConfig.Interfaces = make(map[string]netconf.InterfaceConfig)
+
+	ethNumber := 0
+
+	for _, eth := range m.Interfaces.Public {
+		network := netconf.InterfaceConfig{}
+
+		if eth.IPv4 != nil {
+			network.Gateway = eth.IPv4.Gateway
+
+			network.Addresses = append(network.Addresses, fmt.Sprintf("%s/%s", eth.IPv4.IPAddress, eth.IPv4.Netmask))
+		}
+		if eth.AnchorIPv4 != nil {
+			if strings.HasPrefix(eth.AnchorIPv4.Gateway, "10.") {
```

Comment:
I'd leave it in packet.  I think it's needed because packet sets up a /31 network so you need the larger /8 route for private networking to work. DO doesn't do that same.